pkg/controller/controller_binding.go

```diff
@@ -515,8 +515,11 @@ func (c *controller) transformCredentials(transforms []v1beta1.SecretTransform,
 			}
 			credentials[t.AddKey.Key] = value
 		case t.RenameKey != nil:
-			credentials[t.RenameKey.To] = credentials[t.RenameKey.From]
-			delete(credentials, t.RenameKey.From)
+			value := credentials[t.RenameKey.From]
+			if value != nil {
```

Comment:
I am not sure if this really matters, but could there be a case where the key is present and nil? Perhaps it would be safer to do `value, ok := credentials[t.RenameKey.From]` and test if the key was set, instead of if the value is nil.

Reply:
Thanks. I guess I had my Java hat on when I wrote this if statement. :)